### PR TITLE
LLVM14 Upgrade

### PR DESCRIPTION
**Pull request by yebinchon**
Addressed comments from initial llvm 14 upgrade push, ported fixes from master branch.

### CODE REVIEW

File: lib/MemoryAnalysisModules/FindSource.cpp [hunk 1]
```diff
@@ -105,8 +105,13 @@ const Instruction *findNoAliasSource(const Value *v,
...
     return NULL;
 
+  // FIXME: correctly ported from llvm13
+  // where hasAttribute(unsigned Index, Attribute::AttrKind Kind)
+  // However, why do we need index in the first place?
   if (!f->getAttributes().hasAttributeAtIndex(0, Attribute::NoAlias) &&
-      !isNoAliasCall(v))
+      // isNoAliasFn (deprecated) checked for noAlias attribute & allocation function
+      // isNoAliasCall checks only for noAlias attribute
+      !(isNoAliasCall(v) || isAllocationFn(v, &tli)))
```
Comment (vgene) [inline lib/MemoryAnalysisModules/FindSource.cpp:115]: Are the two cases (get attributes vs is no alias call) the same thing?

File: lib/MemoryAnalysisModules/KillFlow.cpp [hunk 2]
```diff
@@ -950,13 +950,13 @@ LoopAA::ModRefResult KillFlow::modref(const Instruction *i1,
...
       // check whether the pointer of the earlier op (from previous iter) is
       // killed before the load
-      else if (earlierPtr && pointerKilledBefore(L, earlierPtr, load,
-                                                 queryStart, AnalysisTimeout)) {
-        ++numKilledBackwardLoadFlows;
-        LLVM_DEBUG(errs() << "Removed the mod bit at AAA\n");
-        // res = ModRefResult(res & ~Mod);
-        res = NoModRef;
-      }
+      //else if (earlierPtr && pointerKilledBefore(L, earlierPtr, load,
```
Comment (vgene) [inline lib/MemoryAnalysisModules/KillFlow.cpp:954]: What was the commit associate with this?

File: lib/MemoryAnalysisModules/LoopVariantAllocation.cpp [hunk 3]
```diff
@@ -60,7 +60,7 @@ static bool isNoaliasWithinLoop(const Value *src, const Loop *L,
                 0, Attribute::NoAlias))
           return true;
 
-      if (isNoAliasCall(src))
+      if (isNoAliasCall(src) || isAllocationFn(src, &tli))
```
Comment (vgene) [inline lib/MemoryAnalysisModules/LoopVariantAllocation.cpp:66]: Put associate commit here (not sure if this is right)

**Review by vgene — CHANGES_REQUESTED** [review]
Mostly good, a couple of nits

File: lib/MemoryAnalysisModules/FindSource.cpp (hunk 1 above)
Comment (yebinchon) [reply lib/MemoryAnalysisModules/FindSource.cpp:115]: In terms of checking attributes, yes. isNoAliasFn has an additional allocation fcn check, which has been accordingly added.

File: lib/MemoryAnalysisModules/KillFlow.cpp (hunk 2 above)
Comment (yebinchon) [reply lib/MemoryAnalysisModules/KillFlow.cpp:954]: commit 8322cbac18653e72b062f294bcbc49e43349f6d1

File: lib/MemoryAnalysisModules/LoopVariantAllocation.cpp (hunk 3 above)
Comment (yebinchon) [reply lib/MemoryAnalysisModules/LoopVariantAllocation.cpp:66]: Same change as in FindSource.cpp, this is a change I made while upgrading. I can add a FIXME and/or comment.

**Comment by vgene** [comment]
Please add a new commit to inline the discussions here in the source code

**Review by vgene — APPROVED** [review]
LGTM

**Pull request merged**